services/connector/commands/switch_ethereum_chain.go

```diff
@@ -0,0 +1,95 @@
+package commands
+
+import (
+	"database/sql"
+	"errors"
+	"slices"
+
+	persistence "github.com/status-im/status-go/services/connector/database"
+	walletCommon "github.com/status-im/status-go/services/wallet/common"
+)
+
+// errors
+var (
+	ErrNoActiveNetworks     = errors.New("no active networks")
+	ErrUnsupportedNetwork   = errors.New("unsupported network")
+	ErrNoChainIDParamsFound = errors.New("no chain id in params found")
+)
+
+type SwitchEthereumChainCommand struct {
+	NetworkManager NetworkManagerInterface
+	Db             *sql.DB
+}
+
+func (r *RPCRequest) getChainID() (uint64, error) {
+	if r.Params == nil || len(r.Params) == 0 {
+		return 0, ErrEmptyRPCParams
+	}
+
+	switch v := r.Params[0].(type) {
+	case float64:
+		return uint64(v), nil
+	case int:
+		return uint64(v), nil
+	case int64:
+		return uint64(v), nil
```

Comment:
Can you do this like in JS?
```suggestion
	case float64:
	case int:
	case int64:
		return uint64(v), nil
```

Reply:
v is ` interface {}` here, so it should be explicitly separated